### PR TITLE
Adding Unit tests

### PR DESCRIPTION
**Pull request by Fosol**
Adding DAL unit tests
Adding Service unit tests.

### CODE REVIEW

File: backend/Pims.sln [hunk 1]
```diff
@@ -42,6 +40,16 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "docs", "docs", "{A0343C94-4
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "tests", "tests", "{F256F2A5-0DBF-4137-A7D6-21F08111BD4A}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "unit", "unit", "{3D70B211-74A8-484C-9B86-B0A2835C71E7}"
```
Comment (Fosol) [inline backend/Pims.sln:43]: Expanding testing projects.

File: backend/api/Helpers/Extensions/HttpResponseMessageExtensions.cs [hunk 2]
```diff
@@ -48,7 +51,7 @@ public static async Task<IActionResult> HandleResponseAsync<T>(this HttpResponse
                 }
                 else
                 {
-                    var result = new JsonResult(new { Error = error })
+                    var result = new JsonResult(new Models.ErrorResponseModel(error, null))
```
Comment (Fosol) [inline backend/api/Helpers/Extensions/HttpResponseMessageExtensions.cs:54]: Unknown errors will now be wrapped in our standard error format.

File: backend/api/Pims.Api.csproj [hunk 3]
```diff
@@ -19,6 +19,10 @@
     <None Remove="Models\Lookup\**" />
     <None Remove="Profiles\Extensions\**" />
   </ItemGroup>
+  <ItemGroup>
+    <None Remove=".env" />
```
Comment (Fosol) [inline backend/api/Pims.Api.csproj:23]: Shouldn't be included in project file.

File: backend/core/Extensions/CollectionExtensions.cs [hunk 4]
```diff
@@ -18,7 +18,7 @@ public static class CollectionExtensions
         /// <returns></returns>
         public static ICollection<T> RemoveAll<T>(this ICollection<T> items, Func<T, bool> find)
         {
-            var remove = items.Where(find);
+            var remove = items.Where(find).ToArray();
```
Comment (Fosol) [inline backend/core/Extensions/CollectionExtensions.cs:21]: Fixed a bug where the enumeration was changed while iterating.

File: backend/dal/IPimsService.cs [hunk 5]
```diff
@@ -6,7 +6,7 @@ namespace Pims.Dal
     /// <summary>
     /// IPimsService interface, provides a way to interface with the backend datasource.
     /// </summary>
-    public interface IPimsService
+    public interface IPimsService : IService
```
Comment (Fosol) [inline backend/dal/IPimsService.cs:9]: This makes it possible to find all `IService` objects for testing.

File: backend/tests/unit/api/Helpers/ExceptionExtensionsTest.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+using System;
+using Xunit;
+using Entity = Pims.Dal.Entities;
+
+namespace Pims.Api.Test.Helpers
+{
+    [Trait("category", "unit")]
+    [Trait("category", "api")]
+    [Trait("category", "function")]
+    public class ExceptionExtensionsTest
```
Comment (Fosol) [inline backend/tests/unit/api/Helpers/ExceptionExtensionsTest.cs:12]: Finally starting to cover other areas of the code.

File: backend/tests/unit/dal/Services/ParcelServiceTest.cs [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+        public ParcelServiceTest() { }
+        #endregion
+
+        #region Tests
+        #region Get Parcel
+        /// <summary>
+        /// User does not have 'property-view' claim.
+        /// </summary>
+        [Fact]
+        public void Get_NotAuthorized()
```
Comment (Fosol) [inline backend/tests/unit/dal/Services/ParcelServiceTest.cs:28]: Finally testing service layer!

File: backend/dal/IService.cs [hunk 8]
```diff
@@ -0,0 +1,6 @@
+namespace Pims.Dal
+{
+    public interface IService
```
Comment (asanchezr) [inline backend/dal/IService.cs:3]: Please add comments for better intellisense etc

**Pull request merged**